File: JAX-RS/jaxrs-client/src/main/java/edu/academik/jaxrs/client/ResponseApi.java
package edu.academik.jaxrs.client;

import java.io.Serializable;

/**
 *
 * @author Mario Batres
 * @param <T>
 */

public class ResponseApi<T> implements Serializable {

    private boolean success;

    private String message;

    private T content;

    public boolean isSuccess() {
        return success;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getContent() {
        return content;
    }

    public void setContent(T content) {
        this.content = content;
    }

}

File: RetrofitClient/app/src/main/java/edu/academik/ResponseApi.kt
package edu.academik

import com.fasterxml.jackson.annotation.JsonProperty

class ResponseApi<T> {

    @JsonProperty("success")
    var success : Boolean = false

    @JsonProperty("message")
    var message : String? = null

    @JsonProperty("content")
    var content : T? = null
}